internal/metrics/collectors/workqueue.go

```diff
@@ -0,0 +1,117 @@
+package collectors
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/client-go/util/workqueue"
+)
+
+const workqueueSubsystem = "workqueue"
+
+// WorkQueueMetricsCollector implements prometheus.Collector interface
+type WorkQueueMetricsCollector struct {
+	depth        *prometheus.GaugeVec
+	latency      *prometheus.HistogramVec
+	workDuration *prometheus.HistogramVec
+}
+
+// NewWorkQueueMetricsCollector creates a new WorkQueueMetricsCollector
+func NewWorkQueueMetricsCollector(constLabels map[string]string) *WorkQueueMetricsCollector {
+	return &WorkQueueMetricsCollector{
+		depth: prometheus.NewGaugeVec(
+			prometheus.GaugeOpts{
+				Namespace:   metricsNamespace,
+				Subsystem:   workqueueSubsystem,
+				Name:        "depth",
+				Help:        "Current depth of workqueue",
+				ConstLabels: constLabels,
+			},
+			[]string{"name"},
+		),
+		latency: prometheus.NewHistogramVec(
+			prometheus.HistogramOpts{
+				Namespace:   metricsNamespace,
+				Subsystem:   workqueueSubsystem,
+				Name:        "queue_duration_seconds",
```

Comment:
I wonder should this metric and the next one be recorded in milliseconds instead of seconds? I'm just thinking processing might often take less than a second.

Reply:
Yep, that makes sense, thanks.